common/assert-kubeconfig.yml

```diff
@@ -0,0 +1,67 @@
+# checks if kubeconfig pointed to correct cluster 
+- name: "Reading version info"
+  set_fact:
+    cluster_info: "{{ lookup('k8s', cluster_info='version') }}"
+  ignore_errors: true
+
+- set_fact:
+    k8s_version: "{{ cluster_info.kubernetes }}"
+  ignore_errors: true
+
+- set_fact:
+    openshift_version: "{{ cluster_info.openshift }}"
+  ignore_errors: true
+
+- when: openshift_version is not defined
+  ignore_errors: true
+  block:
+  - set_fact:
+      openshift_version_info: "{{ lookup('k8s', 
+        api_version='config.openshift.io/v1', 
+        kind='ClusterVersion',
+        resource_name='version'
+      ) }}"
+  
+  - set_fact:
+      openshift_version: "{{ openshift_version_info.status.desired.version }}"
+    when: openshift_version_info is defined and
+          openshift_version_info.status is defined and
+          openshift_version_info.status.desired is defined and 
+          openshift_version_info.status.desired.version is defined
+
+- when: k8s_version is not defined and
+        openshift_version is not defined
+  fail:
+    msg: "You are not logged into an OpenShift cluster. Exiting..."
+
+- when: openshift_version is not defined
+  set_fact:
+    web_console_url: "{{ lookup('k8s', 
+      resource_name='console', 
+      namespace='openshift-console', 
+      kind='Route',
+      api_verion='route.openshift.io/v1'
+    ) }}"
+  ignore_errors: true
+
+- when: openshift_version is defined or
+        (k8s_version is defined and
+        web_console_url is defined)
+  debug:  
+    msg: |
+      {% if openshift_version is defined and openshift_version.major is defined %}
+      Detected OpenShift version - {{ openshift_version.major|string + '.' + openshift_version.minor|string }}
+      {% elif openshift_version is defined and openshift_version.major is not defined %}
+      Detected OpenShift version - {{ openshift_version|string }}
+      {% elif k8s_version is defined %}
+      Detected k8s version - {{ k8s_version.major|string + '.' + k8s_version.minor|string }}
```

Comment:
would it be too much work to map k8s minor version to openshift 3 version here? 

```
1.7 -> 3.7
1.8 -> 3.8
1.9 -> 3.9
```

would be nice to have for folks who are unaware of the mapping